src/upload/form/mod.rs

```diff
@@ -89,7 +97,7 @@ pub fn build(
                     log::info!("Plain JavaScript project detected. Publishing...");
                     let package_dir = target.package_dir()?;
                     let package = Package::new(&package_dir)?;
-                    let script_path = package.main(&package_dir)?;
+                    let script_path = package_dir.join(package.main(&package_dir)?);
```

Comment:
why the change here?

Reply:
Without it, `script_path` is relative to the entrypoint. By joining `package_dir` with `script_path`, we get an absolute path, which is needed since wrangler actually executes in the project directory (parent).

So yes, it's needed, otherwise wrangler can't find that script to deploy.